tests/core/string/test_node_path.h

```diff
@@ -167,6 +167,53 @@ TEST_CASE("[NodePath] Empty path") {
 			node_path_empty.is_empty(),
 			"The node path should be considered empty.");
 }
+
+TEST_CASE("[NodePath] Slice") {
+	const NodePath node_path_relative = NodePath("Parent/Child:prop");
+	const NodePath node_path_absolute = NodePath("/root/Parent/Child:prop");
+	CHECK_MESSAGE(
+			node_path_relative.slice(0, 2) == NodePath("Parent/Child"),
+			"The slice lower bound should be inclusive and the slice upper bound should be exclusive.");
+	CHECK_MESSAGE(
+			node_path_relative.slice(3) == NodePath(":prop"),
+			"Slicing on the length of the path should return the last entry.");
+	CHECK_MESSAGE(
+			node_path_relative.slice(1, 3) == NodePath("Child:prop"),
+			"Slicing should include names and subnames.");
+	CHECK_MESSAGE(
+			node_path_relative.slice(-1) == NodePath(":prop"),
+			"Slicing on -1 should return the last entry.");
+	CHECK_MESSAGE(
+			node_path_relative.slice(0, -1) == NodePath("Parent/Child"),
+			"Slicing up to -1 should include the second-to-last entry.");
+	CHECK_MESSAGE(
+			node_path_relative.slice(-2, -1) == NodePath("Child"),
+			"Slicing from negative to negative should treat lower bound as inclusive and upper bound as exclusive.");
+	CHECK_MESSAGE(
+			node_path_relative.slice(0, 10) == NodePath("Parent/Child:prop"),
+			"Slicing past the length of the path should work like slicing up to the last entry.");
+	CHECK_MESSAGE(
+			node_path_relative.slice(-10, 2) == NodePath("Parent/Child"),
+			"Slicing negatively past the length of the path should work like slicing from the first entry.");
+	CHECK_MESSAGE(
+			node_path_relative.slice(1, 1) == NodePath(""),
+			"Slicing with a lower bound equal to upper bound should return empty path.");
+
+	CHECK_MESSAGE(
+			node_path_absolute.slice(1, 4) == NodePath("/Parent/Child:prop"),
```

Comment:
I don't think this behavior is ideal. If I take a slice of an absolute path starting from 0, I would expect it to keep the absoluteness, like slicing the test path by `(0, 2)` should be `/root/Parent`. But if the first parameter is greater than zero, it should not be absolute anymore, because `/Parent` does not exist.

Reply:
Yeah, I was wondering about that. It didn't seem to me like there was an obvious advantage to one behavior or the other so I went with the one that was easier to implement. It also begs the question of what should happen with the subnames. For example, should `NodePath("/Parent/Child:prop").slice(-1)` be an absolute path or not? What about `NodePath("Parent/Child:prop").slice(-1)`? It does seem a bit awkward to ever return something like `NodePath("/:prop")`.

I'll go ahead and make your suggested change so that the returned path is only absolute if it includes the first entry of the original path and the original path was also absolute.